### ISSUE TITLE
Turn ResourceTestCase into mixin (to use with TransactionTestCase)

### ISSUE THREAD

**Issue by tonnzor** [django-tastypie/django-tastypie#684]
tastypie provides handy `tastypie.test.ResouceTestCase` class.

However it is inherited from `TestCase` so you cannot use it with `TransactionTestCase`.

Please make it a mixin so you can use these handy methods with any TestCase class you want.

Alternatively, create a separate class `ResouceTransactionTestCase`.

**Comment by jcushman** [comment]
This is still an issue in Django 1.8+. Django provides [a few different test case classes](https://docs.djangoproject.com/en/1.9/topics/testing/tools/#provided-test-case-classes) that are useful for testing APIs, not just TestCase.

I sent a pull req that adds a `ResourceTestCaseMixin` and uses it to implement the existing `ResourceTestCase` as well as a `ResourceTransactionTestCase`.

In the meantime, here's an update of leppert's hack to create a `ResourceTransactionTestCase` for Django 1.8+ and Tastypie 0.12.2:

```
class ResourceTransactionTestCase(TransactionTestCase):
    def setUp(self):
        super(ResourceTransactionTestCase, self).setUp()
        self.serializer = Serializer()
        self.api_client = TestApiClient()
for key, val in ResourceTestCase.__dict__.iteritems():
    if key not in ['setUp', '__doc__']:
        setattr(ResourceTransactionTestCase, key, val)
```

**Comment by SeanHayes** [comment]
Fixed by #1405
